pkg/kn/commands/describe.go

```diff
@@ -0,0 +1,203 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"knative.dev/client/pkg/printers"
+	"knative.dev/pkg/apis"
+)
+
+// Max length When to truncate long strings (when not "all" mode switched on)
+const TruncateAt = 100
+
+func WriteMetadata(dw printers.PrefixWriter, m *metav1.ObjectMeta, printDetails bool) {
+	dw.WriteAttribute("Name", m.Name)
+	dw.WriteAttribute("Namespace", m.Namespace)
+	WriteMapDesc(dw, m.Labels, l("Labels"), "", printDetails)
```

Comment:
I suggest to move this function to `dw`, too (and the remove the `l(...)`).

At least, `l(...)` should be replaced by `printers.Label(...)` (as the whole purpose of this function is to put the label mangling at a single place so that it's easier to add a different label handling (like adding colors).

Reply:
Wait for the next PR...